src/sensai/evaluation/evaluator.py

```diff
@@ -20,10 +21,28 @@
 
 
 class MetricsDictProvider(ABC):
+    def __init__(self):
+        self.trackedExperiment: Optional[TrackedExperiment] = None
+
     @abstractmethod
     def computeMetrics(self, model, **kwargs) -> Dict[str, float]:
         pass
 
+    def setTrackedExperiment(self, trackedExperiment: TrackedExperiment):
```

Comment:
Maybe add a docstring explaining that tracking will take place only if one explicitly calls computeAndTrack rather than computeMetrics, which is perhaps unexpected. (Why aren't we wrapping computeMetrics instead?)

Reply:
I did not want to make too large changes when I included it. Previously experiments were only set in local search loops. I wanted that mechanism to continue being valid, so I did not simply wrap computeMetrics. I guess I could but then the code in hyperopt becomes more convoluted, I think. I will have another look though, these changes were made months ago

Reply:
I meant that the current method `computeMetrics` could become `_computeMetrics` and the new `computeMetrics` could call it and track the results if there is a tracked experiment. It would be completely transparent to `hyperopt`.

Reply:
It wouldn't, since computeMetrics is called in hyperopt which potentially has its own experiment. This way the tracking might happen twice to multiple experiments or can have any kind of strange effects if the same experiment is used in evaluator and hyperopt (I believe for trains it might break). Am I missing something?

Reply:
It's transparent, because it would do what we want without any code changes.
If a user passed the same experiment to both objects, he clearly wants to track stuff twice, doesn't he?

Reply:
Hm, I haven't thought about it like this. I guess we can delegate this to the user, maybe logging a warning or even raising an exception when there is an experiment in both objects.

If we don't raise errors, it might lead to unexpected behaviour, at least with trains. There tracking to the same experiment twice actually overwrites the result. Passing the same experiment to both objects would completely break the tracking and passing two different ones would be nonsensical. I tend to agree with your proposal but let me try it out to see if I'm forgetting something.

Reply:
Ah, I remember: for trains we have to track the result of the full grid search (due to overwriting) and not of a single call of computeMetrics. I wanted to permit setting the tracked experiment in the evaluator and then running the grid search with it, without setting the experiment in the grid search. Then I cannot proceed as you suggested. Do you see this as a valid use case? To me it feels more natural to bind the experiment to the evaluator instead of the hyperopt objects

Reply:
"I wanted to permit setting the tracked experiment in the evaluator and then running the grid search with it, without setting the experiment in the grid search." - that seems contrived to me. The user should set the experiment precisely where we wants stuff to be tracked.

Reply:
Done